va/va.h

```diff
@@ -4182,6 +4182,37 @@ VAStatus vaDeriveImage (
     VAImage *image	/* out */
 );
 
+typedef struct _VAAppSurfaceResInfo{
+    uint32_t image_width;
```

Comment:
This structure declaration looks inconsistent with the rest of the VAAPI. I guess image_height and height are used to specify offset offset between planes, right? But rest of VAAPI, look into VAImage for example, uses offsets[4] for this.

I also think that better if the structure will contain format inside. Comments are needed for some fields, for example for base_addr_align - in which values, bytes, pages, etc.?

I would actually suggest to almost copy declaration of VAImage for this purpose. What you are trying to achieve seems to be almost identical to VAImage structure definition minus allocated data IDs like image_id and buf, plus some alignment hints. It seems something like this:

typedef stuct _VAImageAllocationInfo {
    uint16_t    width;
    uint16_t    height;
    uint32_t    data_size;
    uint32_t    num_planes;     /* can not be greater than 3 */
    uint32_t    pitches[3];
    uint32_t    offsets[3];
    int32_t num_palette_entries;   /* set to zero for non-palette images */
    int32_t entry_bytes;
    int8_t component_order[4];
    uint32_t base_addr_align; /* in bytes, for example =4096 (page alignment) */
    union{
        struct{
        uint32_t islinear  :1;
        uint32_t reserved  :31;
        }bits;
        uint32_t flags;
    };
    uint32_t                va_reserved[VA_PADDING_LOW];
} VAImageAllocationInfo;

Reply:
thanks , accept, but remove     int32_t num_palette_entries;   /* set to zero for non-palette images */
    int32_t entry_bytes;
because the palette format is not used now, to avoid risk , I add more reserved byte , if there are requirement , we can add them back